numpy/lib/npyio.py

```diff
@@ -425,7 +429,7 @@ def save(file, arr, allow_pickle=True, fix_imports=True):
 
     Parameters
     ----------
-    file : file or str
+    file : file, str, or pathlib.Path
         File or filename to which the data is saved.  If file is a file-object,
         then the filename is unchanged.  If file is a string, a ``.npy``
```

Comment:
You could update this to say `If file is a string or a pathlib.Path object, a ''.npy''... extension will be appended to the file name if it does not already...`